src/util/uploads.js

```diff
@@ -0,0 +1,143 @@
+/**
+ * @flow
+ * @file Utility functions for uploads
+ * @author Box
+ */
+
+/**
+ * Returns true if the given object is a Date instance encoding a valid date
+ * (i.e. new Date('this is not a timestamp') should return false).
+ *
+ * Code adapted from
+ * http://stackoverflow.com/questions/1353684/detecting-an-invalid-date-date-instance-in-javascript
+ *
+ * @param {Date} date
+ * @returns {boolean}
+ */
+function isValidDateObject(date: Date) {
+    return Object.prototype.toString.call(date) === '[object Date]' && !isNaN(date.getTime());
+}
+
+/**
+ * Remove milliseconds from date time string
+ * @param {Date} date
+ * @returns {string}
+ */
+function toISOStringNoMS(date: Date) {
+    return date.toISOString().replace(/\.[0-9]{3}/, '');
+}
+
+/**
+ * Returns the file's last modified date as an ISO string with no MS component (e.g.
+ * '2017-04-18T17:14:27Z'), or null if no such date can be extracted from the file object.
+ * (Nothing on the Internet guarantees that the file object has this info.)
+ *
+ * @param {File} file
+ * @returns {?string}
+ */
+function getFileLastModifiedAsISONoMSIfPossible(file: File) {
+    if (isValidDateObject(file.lastModifiedDate)) {
```

Comment:
Deprecated, should not use it
https://developer.mozilla.org/en-US/docs/Web/API/File/lastModifiedDate

Reply:
removed